request.go

```diff
@@ -389,6 +419,78 @@ func (conn *Connection) ExecuteAsync(expr string, args interface{}) *Future {
 	return conn.DoAsync(req)
 }
 
+// PrepareAsync sends a sql statement to prepare and returns PreparedStatement object.
+//
+// Since 1.7.0
+func (conn *Connection) PrepareAsync(expr string) *PreparedStatement {
+	req := newPrepareRequest(expr)
+	fut := conn.DoAsync(req)
+	stmt := newPreparedStatement(fut, conn)
+	return stmt
+}
+
+type PreparedStatementID uint64
+
+// PreparedStatement is a type for handling prepared statements
+//
+// Since 1.7.0
+type PreparedStatement struct {
+	StatementID PreparedStatementID
+	MetaData    []ColumnMetaData
+	ParamCount  uint64
```

Comment:
It may be helpful to have some info about each prepared statement like what count of parameters it may accept or some column metadata with types and so on. Also you have these fields inside the `prepare_table` object in [lua](https://www.tarantool.io/ru/doc/latest/reference/reference_lua/box_sql/prepare/#lua-function.box.prepare) interface. You can also see, that [jdbc](https://developers.google.com/apps-script/reference/jdbc/jdbc-prepared-statement) has some methods that provides this info about the prepared statement.

Reply:
But if an user create a prepared statement then he should know at least `ParamCount`. So it is most likely not needed in the public API.

1. Can the user somehow usefully use a `StatementId`? I think 'yes' because the value can be used (transfered to an another connector as example) and this value cannot be obtained in an other way. 
2. What about `MetaData`? I don't know.

It's easier to make it public later than to make it private. We need to think carefully about how to use it.

Reply:
> We need to think carefully about how to use it.

It is safe to use it.

> But if an user create a prepared statement then he should know at least ParamCount. So it is most likely not needed in the public API.

Not always. This may allow to write some logic that supposes to transfer different PreparedStatement objects between some layers or other objects inside your application. Then user don't need to collect this info manually. I think JDBC supposes such methods for cases like this one.

Reply:
I don't say, that the existing code is consistent with the current API, but it is safe and pretty convenient to use.

Reply:
> Not always. This may allow to write some logic that supposes to transfer different PreparedStatement objects between some layers or other objects inside your application. Then user don't need to collect this info manually. I think JDBC supposes such methods for cases like this one.

But nothing prevents the user from holding info manually (at least `ParamCount`). JDBC it is a Java-style. Redundancy is ok for it.